tests/cli/test_slurm_remote.py

```diff
@@ -0,0 +1,394 @@
+"""Tests that use an actual SLURM cluster.
+
+The cluster to use can be specified by setting the SLURM_CLUSTER environment variable.
+During the CI on GitHub, a small local slurm cluster is setup with a GitHub Action, and
+SLURM_CLUSTER is set to "localhost".
+"""
+from __future__ import annotations
+
+import datetime
+import functools
+import os
+import time
+from logging import getLogger as get_logger
+
+import fabric.runners
+import pytest
+
+from milatools.cli.remote import Remote, SlurmRemote
+
+logger = get_logger(__name__)
+
+SLURM_CLUSTER = os.environ.get("SLURM_CLUSTER")
+JOB_NAME = "milatools_test"
+WCKEY = "milatools_test"
+MAX_JOB_DURATION = datetime.timedelta(seconds=10)
+# BUG: pytest-timeout seems to cause issues with paramiko threads..
+# pytestmark = pytest.mark.timeout(60)
+
+
+_SACCT_UPDATE_DELAY = datetime.timedelta(seconds=10)
+"""How long after salloc/sbatch before we expect to see the job show up in sacct."""
+
+requires_access_to_slurm_cluster = pytest.mark.skipif(
+    not SLURM_CLUSTER,
+    reason="Requires ssh access to a SLURM cluster.",
+)
+
+# TODO: Import the value from `milatools.cli.utils` once the other PR adds it.
+CLUSTERS = ["mila", "narval", "cedar", "beluga", "graham"]
+
+
+@pytest.fixture(scope="session", params=[SLURM_CLUSTER])
+def cluster(request: pytest.FixtureRequest) -> str:
+    """Fixture that gives the hostname of the slurm cluster to use for tests.
+
+    NOTE: The `cluster` can also be parametrized indirectly by tests, for example:
+
+    ```python
+    @pytest.mark.parametrize("cluster", ["mila", "some_cluster"], indirect=True)
+    def test_something(remote: Remote):
+        ...  # here the remote is connected to the cluster specified above!
+    ```
+    """
+    slurm_cluster_hostname = request.param
+
+    if not slurm_cluster_hostname:
+        pytest.skip("Requires ssh access to a SLURM cluster.")
+    return slurm_cluster_hostname
+
+
+def can_run_on_all_clusters():
+    """Makes a given test run on all the clusters in `CLUSTERS`, *for real*!
+
+    NOTE: (@lebrice): Unused here at the moment in the GitHub CI, but locally I'm
+    enabling it sometimes to test stuff on DRAC clusters.
+    """
+    return pytest.mark.parametrize("cluster", CLUSTERS, indirect=True)
```

Comment:
Not sure if the `@lebrice` is still relevant here? I don't see where it is used

Reply:
This isn't used in the CI, but I mark some tests in that file with this, so that they are run on the actual DRAC clusters. I left it here even though it's currently unused because it's very useful and not exactly straightforward to understand or recreate (with the indirect parametrization)